galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/spi/ICertificateStoreService.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright contributors to the Galasa project
+ */
+package dev.galasa.framework.spi;
+
+
+public interface ICertificateStoreService {
+	
+	/**
+	 * <p>
+	 * Retrieves the PEM certificate as a string.
+	 * </p>
+	 * 
+	 * @param id
+	 * @return String certificate
+	 * @throws CertificateStoreException if no certificate found
+	 */
+	public String getPem(String id) throws CertificateStoreException;
+	
+	/**
+	 * <p>
+	 * Retrieves the DER certificate as a string.
+	 * </p>
+	 * 
+	 * @param id
+	 * @return String certificate
+	 * @throws CertificateStoreException if no certificate found
+	 */
+	public String getDer(String id) throws CertificateStoreException;
+	
+	/**
+	 * <p>
+	 * Returns the default keystore. This will be populated by any certificates defined:
+	 * certificates.default.ids=XXXX,YYYY,etc
+	 * certificates.x509.XXXX.pem=...
+	 * certificates.x509.YYYY.der=...
+	 * </p>
+	 * @return
+	 * @throws CertificateStoreException
+	 */
+	public IKeyStore getKeystore() throws CertificateStoreException;
+	
+	/**
+	 * <p>
+	 * Returns a defined group of certificates inside a keystore. This will be populated by any certificates defined:
+	 * certificates.<GroupId>.ids=XXXX,YYYY,etc
+	 * certificates.x509.XXXX.pem=...
+	 * certificates.x509.YYYY.der=...
+	 * </p>
+	 * @param id
+	 * @return
+	 * @throws CertificateStoreException
+	 */
+	public IKeyStore getKeystore(String id) throws CertificateStoreException;
```

Comment:
getKeystoreByGroup(xxxxx)

property needs to be certificates.group.XXXXX.ids
as certificate.MIKE.ids is what?

Reply:
@NotNull